py-polars/polars/functions/lit.py

```diff
@@ -90,7 +90,11 @@ def lit(
         else:
             # value has time zone, but dtype does not: keep value time zone
             if dtype_tz is None:
-                if isinstance(value_tz, ZoneInfo):
+                if isinstance(value_tz, ZoneInfo) or getattr(
+                    value_tz,
+                    "zone",  # handle pytz named timezone.
+                    None,
+                ):
```

Comment:
is it possible to use something like `_check_for_pytz`? (see `_check_for_pyarrow` for a similar function)